starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/tendermint/starport/starport/services/network/networktypes"
+)
+
+type ShowType string
+
+const (
+	chainShowInfo       ShowType = "info"
+	chainShowGenesis    ShowType = "genesis"
+	chainShowAccounts   ShowType = "accounts"
+	chainShowValidators ShowType = "validators"
+	chainShowPeers      ShowType = "peers"
+)
+
+var (
+	showTypes = map[ShowType]struct{}{
+		chainShowInfo:       {},
+		chainShowGenesis:    {},
+		chainShowAccounts:   {},
+		chainShowValidators: {},
+		chainShowPeers:      {},
+	}
+
+	chainGenesisValSummaryHeader = []string{"Genesis Validator", "Self Delegation", "Peer"}
+	chainGenesisAccSummaryHeader = []string{"Genesis Account", "Coins"}
+	chainVestingAccSummaryHeader = []string{"Vesting Account", "Total Balance", "Vesting", "EndTime"}
+)
+
+// NewNetworkChainShow creates a new chain show command to show
+// a chain on SPN.
+func NewNetworkChainShow() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "show [info|genesis|accounts|validators|peers] [launch-id]",
+		Short: "Show details of a chain",
+		RunE:  networkChainShowHandler,
+		Args:  cobra.ExactArgs(2),
+	}
+
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	showType := ShowType(args[0])
+	if _, ok := showTypes[showType]; !ok {
+		cmd.Usage()
+		return fmt.Errorf("invalid arg %s", showType)
+	}
+
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID.
+	launchID, err := network.ParseLaunchID(args[1])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	content := ""
+	switch showType {
+	case chainShowGenesis:
+		content, err = formatChainGenesis(nb, chainLaunch)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), chainLaunch)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowValidators:
+		content, err = formatChainValidators(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(nb NetworkBuilder, chainLaunch networktypes.ChainLaunch) (string, error) {
+	c, err := nb.Chain(networkchain.SourceLaunch(chainLaunch))
+	if err != nil {
+		return "", err
+	}
+
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return "", err
+	}
+	// TODO run the prepare command in dry run mode and print the genesis if not exist
+	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
+		return "", errors.New("chain genesis not found, run the 'network chain prepare' command first")
+	}
+	genesisFile, err := os.ReadFile(genesisPath)
```

Comment:
```suggestion
	genesisFile, err := os.Open(genesisPath)
```

Reply:
The `os.Open` is unnecessary because we are not writing into the file buffer. We only need the file as a read-only. If we are using open, we should also manage the `close` of the file